src/runtime/recipe/particle.ts

```diff
@@ -226,6 +221,29 @@ export class Particle {
   addSlotConnection(name) {
     const slotConn = new SlotConnection(name, this);
     this._consumedSlotConnections[name] = slotConn;
+
+    const slotSpec = this.getSlotSpecByName(name);
+    if (slotSpec) {
+      slotSpec.providedSlots.forEach(providedSlot => {
+        const slot = this.recipe.newSlot(providedSlot.name);
+        slot._sourceConnection = slotConn;
+        slot._name = providedSlot.name; // why do we need to do this when we've provided name to newSlot?
```

Comment:
We will eventually get rid of this code, but it's necessary right now because recipes created by transformation particles can rely on slot id's of provided slots in recipes previously instantiated in the inner arc.